pkg/queryfrontend/roundtrip.go

```diff
@@ -50,7 +52,11 @@ func NewTripperware(config Config, reg prometheus.Registerer, logger log.Logger)
 	queryRangeCodec := NewThanosQueryRangeCodec(config.QueryRangeConfig.PartialResponseStrategy)
 	labelsCodec := NewThanosLabelsCodec(config.LabelsConfig.PartialResponseStrategy, config.DefaultTimeRange)
 
-	queryRangeTripperware, err := newQueryRangeTripperware(config.QueryRangeConfig, queryRangeLimits, queryRangeCodec,
+	queryRangeTripperware, err := newQueryRangeTripperware(
```

Comment:
Is query sharding working with Instant query? I see no reason why it doesn't...

Reply:
Yes, this will be possible, but I think it's better to do it as a follow up.